src/RetryMiddleware.php

```diff
@@ -31,6 +31,7 @@ class RetryMiddleware
         'RequestThrottledException'              => true,
         'TooManyRequestsException'               => true,
         'IDPCommunicationError'                  => true,
+        'InvalidIdentityToken'                   => true,
```

Comment:
Do we want this error code to be retried for all services?

Reply:
[AssumeRoleWithSAML](https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithSAML.html) also returns ```InvalidIdentityToken```.  Other services shouldn't return this error.

Reply:
Ideally retrying `InvalidIdentityToken` would only be limited in scope to STS (as it's a special use case), as it's not inherently a throttling or retryable error, and we should allow for the possibility of other services using that exception name at some point.